physics/cires_ugwp_v1.F90

```diff
@@ -0,0 +1,635 @@
+         j1_tau, j2_tau, ddy_j1tau, ddy_j2tau,                    &
+         j1_qbo, j2_qbo, ddy_j1qbo, ddy_j2qbo, dexpy,             &	 
+         ugrs, vgrs, tgrs, qgrs, prsi, prsl, prslk,               &
+         phii, phil, del, oro_stat,  sgh30, kpbl,                 &
+         dusfcg, dvsfcg, gw_dudt, gw_dvdt, gw_dtdt, gw_kdis,      &
+         tau_tofd, tau_mtb, tau_ogw, tau_ngw,                     &
+         zmtb, zlwb, zogw, du3dt_mtb,du3dt_ogw, du3dt_tms,        &
+         uqbo, ax_qbo, tau_sat, tau_qbo, rdxzb, fhzero, lprnt, ipr)
+	 	 
+!-----------------------------------------------------------
+! Part 1 "revised" LM97 oro-scheme (if do_ugwp=.false.)
+! Part 2  non-stationary multi-wave GWs FV3GFS-v0
+! Part 3  Dissipative version of UGWP-tendency application
+!         (similar to WAM-2017)
+!----------------------------------------------------------- 
+       use machine,          only : kind_phys
+ 
+       use ugwp_common ,     only : rgrav
+       use cires_ugwp_module,only : calendar_ugwp
+       
+       use cires_ugwp_module,only  : vert_qbo, latqbo, widqbo, taurel, kz1, kz2  
+       use cires_ugwp_module, only : ugwp_qbolat
+       use cires_ugwp_module, only : uqboe, u2 => uzmf_merra         
+       use ugwp_wmsdis_init, only  : tamp_mpa, ilaunch
+!   
+       implicit none
+!input
+       real(kind=kind_phys), parameter :: pogw=1.0, pngw=1.0, pked=1.0
+       real(kind=kind_phys), parameter :: fw1_tau=1.0
+        
+       integer, intent(in) :: me,  master, jdat(8)
+       integer, intent(in) :: im, levs, kdt, imx, nmtvr, ipr
+
+       real(kind=kind_phys), intent(in) :: dtp, fhzero
+       real(kind=kind_phys), intent(in) :: cdmbgwd(4)
+
+       logical             :: do_ugwp, do_tofd, lprnt
+       integer, intent(in) :: kpbl(im)
+       real(kind=kind_phys), intent(in), dimension(im) :: xlat, xlatd,   &
+                                 sgh30, sinlat, coslat, spgrid               ! spgrid = tile-area
+     
+       real(kind=kind_phys), intent(in) :: oro_stat(im,nmtvr)
+       
+       real(kind=kind_phys), intent(inout), dimension(im,levs) :: ugrs
+       
+       real(kind=kind_phys), intent(in), dimension(im,levs) ::   &
+                vgrs, tgrs, qgrs, prsi, prsl, prslk, phii, phil, del
+		
+! Temporary changes (removed intent(in))
+       integer, dimension(im) ::j1_tau, j2_tau, j1_qbo, j2_qbo
+       real , dimension(im) :: ddy_j1qbo, ddy_j2qbo, ddy_j2tau, ddy_j1tau 
+       real , dimension(im) :: dexpy(im)     
+!
+!out
+!
+       real(kind=kind_phys), dimension(im,levs) :: gw_dudt, gw_dvdt,    &
+                                                   gw_dTdt, gw_kdis
+
+!-----locals	+ diagnostics output
+
+       real(kind=kind_phys), dimension(im,levs)   :: Pdvdt, Pdudt,      &
+                             Pdtdt, Pkdis, ed_dudt, ed_dvdt, ed_dTdt
+
+       real(kind=kind_phys), dimension(im)      :: dusfcg, dvsfcg
+
+       real(kind=kind_phys), dimension(im)      :: rdxzb, zmtb,          &
```

Comment:
Usual checks: all arrays in the argument list that may not be allocated (depending on some runtime flags) need to have assumed size declarations
```
real(kind=kind_phys), :: foo(:)
real(kind=kind_phys), :: bar(:,:)
```